qiskit/result/utils.py

```diff
@@ -127,14 +126,60 @@ def _adjust_creg_sizes(creg_sizes, indices):
     return new_creg_sizes
 
 
+def marginal_memory(
+    memory: List[str],
+    indices: Optional[List[int]] = None,
+    int_return: bool = False,
```

Comment:
This is just a curiosity. Is the list of integer more efficient in memory footprint than binary ndarray? Given we use memory information to run restless analysis in qiskit experiment, it should take memory efficient representation to run a parallel experiment in 100Q+ device.

Reply:
Do you mean like storing the shot memory as a 2d array where each row has n elements for each bit or something else? The list of ints here will be more memory efficient than that in the rust side because I'm using a `Vec<BigUint>` (whch is just a Vec of digits internally) and it will not be fixed with for each shot. The python side I expect would be similar since the Python integer class is very similar to BigUint (a byte array of digits). (although list isn't necessarily as compact as a `Vec<T>`/`ndarray`). I think it would be best to test this though to be sure and settle on a common way to represent large results values in a non-string type.

As an aside I only used a list here because numpy doesn't have support for arbitrary large integers (outside of using a `object` dtype, which ends up just being a pointer to the python heap, for python ints) and I was worried about the

Reply:
Thanks. Sounds like current implementation is reasonable.